metarecord/views/function.py

```diff
@@ -295,6 +297,31 @@ def get_queryset(self):
 
         return queryset.latest_version()
 
+    def retrieve(self, request, *args, **kwargs):
+        if not validate_uuid4(self.kwargs.get('uuid')):
+            raise exceptions.ValidationError(_('Invalid UUID'))
+
+        try:
+            instance = self.get_object()
+        except (Function.DoesNotExist, Http404):
+            instance = None
+
+        if not instance:
+            filter_kwargs = {self.lookup_field: self.kwargs[self.lookup_field]}
+
+            if 'version' in self.request.query_params:
+                filter_kwargs = {**filter_kwargs, 'version': self.request.query_params['version']}
+
+            qs = Function.objects.filter(**filter_kwargs)
+
+            if qs.exists():
+                raise exceptions.NotAuthenticated
+
+            raise exceptions.NotFound
```

Comment:
As this part probably seems strange for people not knowing the project well (for example you would have to know that for unauthenticated users other versions are already filtered out elsewhere) I would like this to have a comment that briefly explains the logic and the need for it.

Reply:
Added small comment that briefly explains reasoning behind this.